tests/view_tests/tests/test_debug.py

```diff
@@ -1081,9 +1084,50 @@ def __repr__(self):
         reporter = ExceptionReporter(None, exc_type, exc_value, tb)
         html = reporter.get_traceback_html()
         self.assertEqual(len(html) // 1024 // 128, 0)  # still fit in 128Kb
-        self.assertIn(
-            "&lt;trimmed %d bytes string&gt;" % (large + repr_of_str_adds,), html
+        trim_msg = "&lt;trimmed %d bytes string&gt;" % (
+            large - EXCEPTION_PRINT_LIMIT + repr_of_str_adds,
         )
+        self.assertIn(trim_msg, html)
+
+    def test_large_sizable_object(self):
+        """Large objects should not be rendered entirely"""
+        lg = list(range(1000 * 1000))
+        try:
+            lg["a"]
+        except TypeError:
+            exc_type, exc_value, tb = sys.exc_info()
+
+        reporter = ExceptionReporter(None, exc_type, exc_value, tb)
+        d = reporter.get_traceback_data()
+        vars = d["lastframe"]["vars"]
+
+        for k, v in vars:
```

Comment:
Given that we don't (can't!) test Python, we assume Python and its libraries to be correct. So we can certainly use reprlib in the tests to assert our desired results.

What we should test is that our choosing of max limits and related are correct. In fact, we should most definitely provide tests for our custom `Repr` class. Once we ensure that our `Repr` class is correct, we could also use it to write these tests.

Does that make sense? (I can expand if not, no worries, let me know)

Reply:
Adding tests to check the custom class on common data structures maybe?